packages/monaco/src/browser/monaco-editor-provider.ts

```diff
@@ -101,6 +105,17 @@ export class MonacoEditorProvider {
         const options = this.createMonacoEditorOptions(model);
         const editor = new MonacoEditor(uri, model, document.createElement('div'), this.m2p, this.p2m, options, override);
         toDispose.push(this.editorPreferences.onPreferenceChanged(event => this.updateMonacoEditorOptions(editor, event)));
+
+        editor.document.onWillSaveModel(event => {
+            event.waitUntil(new Promise<monaco.editor.IIdentifiedSingleEditOperation[]>(async resolve => {
+                // event.reason = 2: TextDocumentSaveReason.AfterDelay means: autoSave
+                if (event.reason !== 2 && this.editorPreferences['editor.formatOnSave']) {
```

Comment:
```ts
event.reason !== TextDocumentSaveReason.AfterDelay
```

Reply:
Done for the next version, will use event.reason === TextDocumentSaveReason.Manual